tmt/steps/prepare/feature.py

```diff
@@ -0,0 +1,356 @@
+import dataclasses
+import enum
+from typing import Optional
+
+import tmt
+import tmt.base
+import tmt.log
+import tmt.options
+import tmt.steps
+import tmt.steps.prepare
+import tmt.utils
+from tmt.steps.provision import Guest
+from tmt.utils import ShellScript, field
+
+
+class Distro(enum.Enum):
+    FEDORA = enum.auto()
+    RHEL_7 = enum.auto()
+    RHEL_8 = enum.auto()
+    RHEL_9 = enum.auto()
+    CENTOS_7 = enum.auto()
+    CENTOS_STREAM_8 = enum.auto()
+    CENTOS_STREAM_9 = enum.auto()
+
+
+class Feature(tmt.utils.Common):
+    """ Base class for feature implementations """
+
+    KEY: str
+
+    def __init__(
+            self,
+            *,
+            parent: 'PrepareFeature',
+            guest: Guest,
+            logger: tmt.log.Logger) -> None:
+        """ Initialize feature data """
+        super().__init__(logger=logger, parent=parent, relative_indent=0)
+        self.guest = guest
+        self.logger = logger
+        self.guest_sudo = self.get_guest_sudo()
+        self.guest_distro = self.get_guest_distro()
+
+    def get_guest_sudo(self) -> str:
+        """ Return 'sudo' if guest is not superuser """
+        if not self.guest.facts.is_superuser:
+            return 'sudo'
+        return ''
+
+    def get_guest_distro(self) -> Optional[Distro]:
+        """ Get guest distro by parsing the guest facts """
+        os_release = self.guest.facts.os_release_content
+        if os_release is None:
+            return None
+
+        if os_release.get('NAME') == 'Fedora Linux':
+            return Distro.FEDORA
+
+        if os_release.get('NAME') == 'Red Hat Enterprise Linux':
+            if os_release.get('VERSION_ID', '').startswith('9'):
```

Comment:
maybe rather startswith `9.`, while I do not believe we will live until RHEL-71 and simlar. but maybe tmt will :dancers:

Reply:
Updated to match `n.` where n is an integer. e.g.

```python
if os_release.get('VERSION_ID', '').startswith('9.'):
```